### ISSUE TITLE
Your opinion about iOS/MacOS development foreground prospect

### ISSUE THREAD

**Issue by jamesdouble** [mxcl/AmA#2]
Hi, I am a Chinese iOS Developer.

iOS Develop have been publish almost ten year, 
and I think iOS Native Develop is not so popular and less market demanding like before.

So I would like ask your opinion about iOS/MacOS development's future trending?

**Comment by mxcl** [comment]
Yeah I agree that iOS has peaked and macOS was always niche, especially in this era of cross platform Javascript.

However iOS is still an enormous platform, and it's the platform that is usually still targeted: first, and for the “flagship” app for that company. Also in my experience for developing for both Android and iPhone, it is less tedious to develop for.

However in the last 10 years the sheer number of new people learning iOS and entering the industry has been astonishing.

It’s hard to ignore the Javascript trend though, and if employability is your main concern then I would gain experience there. One thing I always say about the tech industry is that it is always changing and you must keep an eye on the horizon. Once upon a time knowing Flash made you the highest paid developer around, and then rapidly, within a couple of years it was dead.

Having said all this I believe in the future of Swift as a cross platform language (for apps and other tools) and personally I am actively working on tooling in that front that I plan to release soon.

Another thing to consider: 10 years is all the time there is for iOS, so you are in the top 1% of iOS developers.
